client/components/email-verification/email-verification-dialog/index.jsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import noop from 'lodash/noop';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { get, includes } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Dialog from 'components/dialog';
+import Button from 'components/button';
+import Spinner from 'components/spinner';
+import { getCurrentUserEmail } from 'state/current-user/selectors';
+import {
+	verifyEmail,
+	resetVerifyEmailState,
+} from 'state/current-user/email-verification/actions';
+
+class VerifyEmailDialog extends Component {
+
+	getResendButtonLabel() {
+		if ( 'sent' === this.props.emailVerification ||
+			'error' === this.props.emailVerification ) {
+			return this.props.translate( 'Email Sent' );
+		}
+		if ( 'requesting' === this.props.emailVerification ) {
+			return <Spinner className="email-verification-dialog__confirmation-dialog-spinner" />;
+		}
+		return this.props.translate( 'Resend Email' );
+	}
+
+	handleClose = () => {
+		this.props.resetVerifyEmailState();
+		this.props.onClose();
+	};
+
+	getDialogButtons() {
+		return [
+			<Button
+				key="resend"
+				primary={ false }
+				disabled={ includes( [ 'requesting', 'sent', 'error' ], this.props.emailVerification ) }
```

Comment:
It's interesting that we only get one chance to resend, but it looks like this was previous behavior,